### PR TITLE
Add C package xsel recipe

### PR DESCRIPTION
**Pull request by ehfd**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ehfd** [comment]
@conda-forge/help-c-cpp Ready for review.

This package interfaces with the X11 clipboard API.

Since this is not an official X.Org package, keeping Unix-only for now.

**Comment by ehfd** [comment]
@hmaarrfk
I have seven X11-related packages waiting to be merged. Could you possibly help out?

**Comment by ehfd** [comment]
@xhochy Very thankful if these could be processed in timely fashion:

https://github.com/conda-forge/staged-recipes/pull/26163
https://github.com/conda-forge/staged-recipes/pull/26165
https://github.com/conda-forge/staged-recipes/pull/26166
https://github.com/conda-forge/staged-recipes/pull/26242
https://github.com/conda-forge/staged-recipes/pull/26254
https://github.com/conda-forge/staged-recipes/pull/26256
https://github.com/conda-forge/staged-recipes/pull/26333

I am waiting for this to do a release on my own project.

File: recipes/xsel/meta.yaml
Comment (hmaarrfk) [inline recipes/xsel/meta.yaml:29]: shouldn't this be in the host section?

Comment (ehfd) [reply recipes/xsel/meta.yaml:29]: They're build dependencies as per Homebrew and Arch.

Comment (ehfd) [reply recipes/xsel/meta.yaml:29]: https://formulae.brew.sh/formula/xsel

https://archlinux.org/packages/extra/x86_64/xsel/

**Comment by ehfd** [comment]
> @xhochy Very thankful if these could be processed in timely fashion:
> 
> #26163 #26165 #26166 #26242 #26254 #26256 #26333
> 
> I am waiting for this to do a release on my own project.

@h-vetinari Since you're in the core team, could you help this out? It's been over a month.

**Comment by ehfd** [comment]
This is an optional dependency of (indirectly) `pandas` and (directly) `pyperclip`.

File: recipes/xsel/meta.yaml
Comment (h-vetinari) [reply recipes/xsel/meta.yaml:29]: Even if it just needs the headers, it should go into host. Different tools/ecosystems don't necessarily agree on all details of naming/semantics, but in conda-forge, the build environment is restricted purely to things that need to be executed during the build (like a compiler); everything you want to compile _against_, whether shared, static or header-only, needs to go into host.

Comment (ehfd) [reply recipes/xsel/meta.yaml]: Moved to headers and instead put in ignore_run_export then.

Comment (ehfd) [reply recipes/xsel/meta.yaml]: @h-vetinari

File: recipes/xsel/build.sh
Comment (h-vetinari) [inline recipes/xsel/build.sh:17]: You already have `set -x` at the top, so all these echos are redundant.

File: recipes/xsel/meta.yaml
Comment (h-vetinari) [inline recipes/xsel/meta.yaml:3]: I don't see this used anywhere in `build.sh`

File: recipes/xsel/meta.yaml
Comment (h-vetinari) [inline recipes/xsel/meta.yaml:6]: I prefer not to template the name; not like it could change anyway (also remove first line)
```suggestion
  name: xsel
```

**Review by h-vetinari — COMMENTED** [review]
Basically LGTM, just some small details to fix up

File: recipes/xsel/build.sh
Comment (ehfd) [reply recipes/xsel/build.sh]: Done.

File: recipes/xsel/meta.yaml
Comment (ehfd) [reply recipes/xsel/meta.yaml]: Done.

File: recipes/xsel/meta.yaml
Comment (ehfd) [reply recipes/xsel/meta.yaml]: Done.

File: recipes/xsel/build.sh
Comment (h-vetinari) [inline recipes/xsel/build.sh:39]: Now that you deleted the initial comment about the libtool files, this has lost its relevant context & reference. Can you please re-add that link? The rest of the changes looks like a nice clean-up! 👍

Comment (ehfd) [reply recipes/xsel/build.sh:45]: Completed. And I've put in your suggestions to the other six recipes as well.

Comment (ehfd) [reply recipes/xsel/build.sh:45]: https://github.com/conda-forge/staged-recipes/pull/26163 https://github.com/conda-forge/staged-recipes/pull/26165 https://github.com/conda-forge/staged-recipes/pull/26166 https://github.com/conda-forge/staged-recipes/pull/26242 https://github.com/conda-forge/staged-recipes/pull/26254 https://github.com/conda-forge/staged-recipes/pull/26256 https://github.com/conda-forge/staged-recipes/pull/26333

I've synced your suggestions to the other recipes.

**Review by h-vetinari — APPROVED** [review]
LGTM!

**Comment by ehfd** [comment]
@h-vetinari You could merge if you want.

**Comment by hmaarrfk** [comment]
ehfd  are you trying to also build packages for OSX and aarch? 

Maybe we should add:

```
provider:
  linux_aarch64: default
build_platform:
  linux_aarch64: linux_64
  linux_ppc64le: linux_64
  osx_arm64: osx_64
conda_build:
  pkg_format: '2'
conda_forge_output_validation: true
test: native_and_emulated
```

in a file `conda-forge.yml` to build them on the first run optimistically.

**Comment by hmaarrfk** [comment]
similar to https://github.com/conda-forge/staged-recipes/pull/25992/files#diff-b06c1773820af0e7c38a9e9c53a478b75afd90b056775bd3c4c8cd950c6b8276 (but with ppc64le)

**Comment by ehfd** [comment]
@hmaarrfk I agree because I had to add them in evdev after the feedstock got created. Just a second.

**Comment by ehfd** [comment]
```
build_platform:
  linux_aarch64: linux_64
  linux_ppc64le: linux_64
  osx_arm64: osx_64
conda_build:
  error_overlinking: true
conda_forge_output_validation: true
github:
  branch_name: main
  tooling_branch_name: main
```

This seems to be the most recent version (default from a few days ago). Added.

File: conda-forge.yml
Comment (hmaarrfk) [inline conda-forge.yml:10]: you should not edit this feedstock's file.

rather you need to create this file within the directory

`recipes/xsel`

Comment (ehfd) [reply conda-forge.yml:10]: Oh. OK.

Comment (ehfd) [reply conda-forge.yml:10]: That was what @xhochy said to me earlier.

Comment (ehfd) [reply conda-forge.yml]: Fixed.

Comment (hmaarrfk) [reply conda-forge.yml]: lets see, its kinda a gamble if it works, but you are trying to get quite a few feedstocks in, so it might be helpful

File: recipes/xsel/conda-forge.yml
Comment (hmaarrfk) [inline recipes/xsel/conda-forge.yml:10]:
```suggestion
  tooling_branch_name: main
conda_build:
  pkg_format: '2'
conda_forge_output_validation: true
test: native_and_emulated

```

Comment (ehfd) [reply recipes/xsel/conda-forge.yml:10]: I thought that `pkg_format: '2'` at least is the default?

Comment (ehfd) [reply recipes/xsel/conda-forge.yml:10]: Anyways, applied.

Comment (ehfd) [reply recipes/xsel/conda-forge.yml:10]: Everything has been synced to all recipes as adequate. @hmaarrfk

Comment (ehfd) [reply recipes/xsel/conda-forge.yml:10]: It is with remorse that the "gamble" has no effect. I'll change the arches manually.

@hmaarrfk

**Pull request merged**
